samples/SemanticKernelAgent/SemanticKernelTravelAgent.cs

```diff
@@ -0,0 +1,572 @@
+        }
+    }
+
+    /// <summary>
+    /// Checks if the HTTP response indicates a transient error
+    /// </summary>
+    /// <param name="response">HTTP response message</param>
+    /// <returns>True if the status code indicates a transient error</returns>
+    private bool IsTransientError(HttpResponseMessage response)
+    {
+        int statusCode = (int)response.StatusCode;
+        return statusCode == 408 // Request Timeout
+            || statusCode == 429 // Too Many Requests 
+            || statusCode >= 500 && statusCode < 600; // Server errors
+    }
+}
+#endregion
+
+#region Response Format
+/// <summary>
+/// A Response Format model to direct how the model should respond
+/// </summary>
+public class ResponseFormat
+{
+    [JsonPropertyName("status")]
+    public string Status { get; set; } = "input_required";
+
+    [JsonPropertyName("message")]
+    public string Message { get; set; } = "";
+
+    public static ResponseFormat FromJson(string json)
+    {
+        try
+        {
+            return JsonSerializer.Deserialize<ResponseFormat>(json) ??
+                   new ResponseFormat { Status = "error", Message = "Failed to parse response" };
+        }
+        catch
+        {
+            return new ResponseFormat { Status = "error", Message = "Invalid response format" };
+        }
+    }
+}
+#endregion
+
+#region Semantic Kernel Agent
+
+/// <summary>
+/// Wraps Semantic Kernel-based agents to handle Travel related tasks
+/// </summary>
+public class SemanticKernelTravelAgent : IDisposable
+{
+    public static readonly ActivitySource ActivitySource = new ActivitySource("A2A.SemanticKernelTravelAgent", "1.0.0");
+
+    /// <summary>
+    /// Initializes a new instance of the SemanticKernelTravelAgent
+    /// </summary>
+    /// <param name="configuration">Application configuration</param>
+    /// <param name="logger">Logger for the agent</param>
+    /// <param name="httpClient">HTTP client</param>
+    public SemanticKernelTravelAgent(
+        IConfiguration configuration,
+        HttpClient httpClient,
+        ILogger logger)
+    {
+        _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+        _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+
+        // Create currency plugin
+        _currencyPlugin = new CurrencyPlugin(
+            logger: new Logger<CurrencyPlugin>(new LoggerFactory()),
+            httpClient: _httpClient);
+
+        // Initialize the agent
+        _agent = InitializeAgent();
+    }
+
+    /// <summary>
+    /// Dispose of resources
+    /// </summary>
+    public void Dispose()
+    {
+        GC.SuppressFinalize(this);
+    }
+
+    public void Attach(TaskManager taskManager)
+    {
+        this._taskManager = taskManager;
+        taskManager.OnTaskCreated = async (task) =>
```

Comment:
These lambda shouldn't be necessary as the signature of ExecuteAgentTask matches the function